src/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -97,11 +100,13 @@
     BAMFileReader(final InputStream stream,
                   final File indexFile,
                   final boolean eagerDecode,
+                  final boolean useAsynchronousIO,
```

Comment:
Should you add `setUseAsyncIO()` to `SamReader.ReaderImplementation`, which already has methods like `enableIndexCaching()`, instead of adding a new constructor argument?